django/core/management/commands/loaddata.py

```diff
@@ -84,6 +84,31 @@ def handle(self, *fixture_labels, **options):
         if transaction.get_autocommit(self.using):
             connections[self.using].close()
 
+    @cached_property
+    def compression_formats(self):
+        # Forcing binary mode may be revisited after dropping Python 2 support (see #22399)
+        compression_formats = {
+            None: (open, 'rb'),
+            'gz': (gzip.GzipFile, 'rb'),
+            'zip': (SingleZipReader, 'r'),
+            'stdin': (lambda *args: sys.stdin, None),
+        }
+        if has_bz2:
+            compression_formats['bz2'] = (bz2.BZ2File, 'r')
+        if has_lzma:
+            compression_formats['lzma'] = (lzma.LZMAFile, 'r')
+            compression_formats['xz'] = (lzma.LZMAFile, 'r')
+        return compression_formats
+
+    def _reset_sequences(self, connection):
```

Comment:
Does this need a short explanation now that it's factored out?

Reply:
463566e467ebf21e9feffeb8c8d4c62a7d6e80c2